VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -187,7 +186,17 @@ public static CharSequence trim(CharSequence s) {
     }
 
     public static CharSequence removeDuplicateWhitespace(CharSequence text) {
-        return PATTERN_DUPLICATE_WHITESPACE.matcher(text).replaceAll(" ");
+        SpannableStringBuilder builder = new SpannableStringBuilder(text);
+        for (int size = text.length(), i = 0; i < size; i++) {
+            if (Character.isWhitespace(builder.charAt(i))) {
```

Comment:
I think we can just check for standard whitespace character `' '` here, instead of every unicode whitespace character, as we should only be using the standard one in the strings, and we're replacing them with it anyway.